api/src/main/java/org/pmiops/workbench/firecloud/FireCloudConfig.java

```diff
@@ -128,13 +127,13 @@ public StatusApi statusApi(WorkbenchConfig workbenchConfig) {
     ApiClient apiClient = new ApiClient();
     apiClient.setBasePath(workbenchConfig.firecloud.baseUrl);
     apiClient.setDebugging(workbenchConfig.firecloud.debugEndpoints);
-    addFireCloudDefaultHeader(apiClient);
+    addFireCloudDefaultHeader(apiClient, workbenchConfig.firecloud.xAppIdValue);
     statusApi.setApiClient(apiClient);
     return statusApi;
   }
 
-  private void addFireCloudDefaultHeader(ApiClient apiClient) {
-    apiClient.addDefaultHeader(X_APP_ID_HEADER, X_APP_ID_HEADER_VALUE);
+  private void addFireCloudDefaultHeader(ApiClient apiClient, String value) {
```

Comment:
This method feels a bit awkward, given it has only a single line and takes a parameter derived from the workbenchConfig, which this class already has access to.

Idea: maybe refactor this into a buildApiClient() method which handles as much common ApiClient logic as possible:

ApiClient apiClient = new ApiClient();
apiClient.setBasePath(workbenchConfig.firecloud.baseUrl);
apiClient.setDebugging(workbenchConfig.firecloud.debugEndpoints);
apiClient.addDefaultHeader(X_APP_ID_HEADER, workbenchConfig.firecloud.xAppIdValue);
return apiClient;

If you make that method public, we could even call it from FireCloudServiceImpl to remove a few lines of code from there too.